src/EditorFeatures/Core/ValueTracking/SerializableValueTrackedItem.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Remote;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.ValueTracking
+{
+    [DataContract]
+    internal class SerializableValueTrackedItem : IEquatable<SerializableValueTrackedItem>
```

Comment:
do you need this to be equatable?  do you expect to put these in collections?

Reply:
consider a readonly struct here.

Reply:
> do you need this to be equatable? do you expect to put these in collections?

They are going to be in immutable arrays at the very least, but maybe don't need to be equatable? I wasn't sure if there was some requirement for serialization with that.

Reply:
sorry, i meant: put in a dictionary.

If you're nto going to put into that (or otherwise call .Equals or .GetHashCode yourself) you don't need to implement this.  

>  I wasn't sure if there was some requirement for serialization with that.

Nope, and in general  i recommend making them just plain POCOs.

In FAR some things are equatable because we do put htem in dictionaries.  for example, we often say "here's a serialized definition" and then "here's a serialized reference associated with this serialized definition".  we need to match those two definitions up so that as we deserialize the deserialized reference gets assocaited with the righ tdeserialized definition (i.e. preserving reference identity once deserialized).  if you don't hae that need, you don't need this.  

You'll know if you have that need though if your serialization logic needs to use a dictionary/set to ensure this sort of invariant.